source/distributed/grid_refinement.cc

```diff
@@ -264,12 +264,13 @@ namespace internal
                                   return c > test_threshold;
                                 });
 
-                unsigned int total_count = 0;
```

Comment:
Oh, does the MPI call actually work if input and output variable have different types? `my_count` is still `unsigned int`.

Reply:
no, you are right. This is a problem.